sdk/eventhubs/azure_messaging_eventhubs/src/error.rs

```diff
@@ -61,21 +57,11 @@ impl std::error::Error for EventHubsError {
 impl std::fmt::Display for EventHubsError {
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
         match &self.kind {
-            ErrorKind::UnableToAddConnection => f.write_str("Unable to add connection."),
-            ErrorKind::MissingMessageSender => f.write_str("Missing message sender."),
-            ErrorKind::MissingMessageReceiver => f.write_str("Missing message receiver."),
+            ErrorKind::SimpleMessage(msg) => write!(f, "{}", msg),
+            ErrorKind::AzureCore(e) => write!(f, "Azure Core Error: {}", e),
             ErrorKind::SendRejected(e) => write!(f, "Send rejected: {:?}", e),
             ErrorKind::InvalidManagementResponse => f.write_str("Invalid management response"),
-            ErrorKind::UnableToAddAuthenticationToken => {
-                f.write_str("Unable to add authentication token")
-            }
-            ErrorKind::MissingSession => {
-                f.write_str("The session for the specified partition is missing.")
-            }
             ErrorKind::AmqpError(source) => write!(f, "AmqpError: {:?}", source),
```

Comment:
nit, we say "Azure Core Error" above and not "AzureCore", so for consistency I suggest
```suggestion
            ErrorKind::AmqpError(source) => write!(f, "AMQP Error: {:?}", source),
```